test/e2e/bgppolicy_test.go

```diff
@@ -123,84 +114,181 @@ func TestBGPPolicy(t *testing.T) {
+		t.Log("Updating the BGP configuration on the remote FRR router")
+		configureExternalBGPRouter(t, int32(65000), int32(64513), false)
+
+		t.Log("Updating InternalTrafficPolicy of the test Service from Cluster to Local")
+		_, err = data.updateServiceInternalTrafficPolicy("agnhost-svc", true)
 		require.NoError(t, err)
-		require.Equal(t, 0, rc)
-		require.Equal(t, podName, stdout)
-	}
+		defer data.updateServiceInternalTrafficPolicy("agnhost-svc", false)
+
+		t.Logf("Updating the test BGPPolicy %q: setting a new local ASN and removing Pod CIDR advertisement", bgpPolicy.Name)
+		updatedBGPPolicy := bgpPolicy.DeepCopy()
+		updatedBGPPolicy.Spec.LocalASN = int32(64513)
+		updatedBGPPolicy.Spec.Advertisements.Pod = nil
+		_, err = data.CRDClient.CrdV1alpha1().BGPPolicies().Update(context.TODO(), updatedBGPPolicy, metav1.UpdateOptions{})
+		require.NoError(t, err)
+
+		t.Log("Getting routes installed on the remote FRR router and verifying them")
+		expectedRoutes = []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: []string{nodeIPv4(0)}}}
+		notExpectedRoutes := []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()}}
+		checkFRRRouterBGPRoutes(t, expectedRoutes, notExpectedRoutes)
+
+		t.Log("Verifying the connectivity of the installed routes on the remote FRR route")
+		ipsToConnect = []string{clusterIP}
+		for _, ip := range ipsToConnect {
+			checkFRRRouterBGPRouteConnectivity(ip)
+		}
+	})
+
+	t.Run("Multiple BGPPolicies applied to different Nodes within a confederation", func(t *testing.T) {
+		t.Log("Configuring the remote FRR router with BGP")
+
+		asnStart := int32(60000)
+		confederationID1 := int32(64512)
+		confederationID2 := int32(64513)
+
+		configureExternalBGPRouter(t, int32(65000), confederationID1, true)
+
+		var bgpPolicies []*crdv1alpha1.BGPPolicy
+		for i := 0; i < len(clusterInfo.nodes); i++ {
+			bgpPolicyName := "test-policy-" + strconv.Itoa(i)
+			localASN := asnStart + int32(i)
+			bgpPolicy := &crdv1alpha1.BGPPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: bgpPolicyName,
+				},
+				Spec: crdv1alpha1.BGPPolicySpec{
+					NodeSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							labelNodeHostname: clusterInfo.nodes[i].name,
+						},
+					},
+					LocalASN:   localASN,
+					ListenPort: ptr.To[int32](179),
+					Confederation: &crdv1alpha1.Confederation{
+						Identifier: confederationID1,
+					},
+					Advertisements: crdv1alpha1.Advertisements{
+						Service: &crdv1alpha1.ServiceAdvertisement{
+							IPTypes: []crdv1alpha1.ServiceIPType{crdv1alpha1.ServiceIPTypeClusterIP},
+						},
+						Pod: &crdv1alpha1.PodAdvertisement{},
+					},
+					BGPPeers: []crdv1alpha1.BGPPeer{
+						{Address: externalInfo.externalFRRIPv4, ASN: int32(65000)},
+					},
+				},
+			}
+			t.Logf("Creating a test BGPPolicy %q with ASN %d, the confederation ID 65000, applied to Node %s", bgpPolicy.Name, localASN, clusterInfo.nodes[i].name)
```

Comment:
I don't think 65000 is the confederation ID?

Reply:
Done